src/xngin/apiserver/routers/experiments/experiments_api.py

```diff
@@ -139,6 +141,47 @@ async def get_assignment_for_participant_with_apikey(
         experiment=experiment,
         participant_id=participant_id,
         create_if_none=create_if_none,
+        properties=None,
+        random_state=random_state,
+    )
+
+
+@router.post(
+    "/experiments/{experiment_id}/assignments/{participant_id}/assign_with_filters",
```

Comment:
@poorva-noora this is the proposed endpoint for online assignments; please also see OnlineAssignmentWithFiltersRequest. Should be pretty straightforward, but let me know if you have concerns about the simple payload to POST, thanks!